### PR TITLE
[CHK-1386] feat: add integration test stage for deploy pipeline

### PR DESCRIPTION
**Pull request by MarcoAGreco**
This PR update the devops deploy pipeline adding two new steps of API tests to test the integration of the deployed microservice in the eCommerce environment. The stages test the development deployment, the blue UAT deployment and the green UAT deployment. 

#### List of Changes

- Add API integration tests in deploy pipeline

#### Motivation and Context

- Increase code quality
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .devops/azure-templates/api-tests.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+
+steps:
+  - script: |
+      yarn global add newman
+    displayName: 'Newman installation'
+  - script: |
+      newman run api-tests/checkout-tests/checkout-for-ecommerce-api.tests.json -e ${{ parameters.ENV_FILE }} --ignore-redirects --reporters cli,junit --reporter-junit-export Results/api-TEST.xml
+    displayName: 'Run api test app - env file: ${{ parameters.ENV_FILE }}'
+    continueOnError: true
```
Comment (giovanniberti) [inline .devops/azure-templates/api-tests.yml:14]: See https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/189#discussion_r1165098714

**Review by giovanniberti — CHANGES_REQUESTED** [review]

File: .devops/azure-templates/api-tests.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  - name: "ENV_FILE"
+    type: string
+
+
+steps:
+  - script: |
+      yarn global add newman
+    displayName: 'Newman installation'
+  - script: |
+      newman run api-tests/checkout-tests/checkout-for-ecommerce-api.tests.json -e ${{ parameters.ENV_FILE }} --ignore-redirects --reporters cli,junit --reporter-junit-export Results/api-TEST.xml
```
Comment (pietro-tota) [inline .devops/azure-templates/api-tests.yml:12]: differentiate output results file name based on env file (maybe interpolating env file name)

**Review by pietro-tota — APPROVED** [review]
Just a comment

**Comment by infantesimone** [comment]
to merge @MarcoAGreco

**Pull request merged**